File: src/015/p015.go
package main

import "fmt"

func combination(n, k int) int {
	c := 1
	for i := 0; i < k; i++ {
		c *= n - i
		c /= i + 1
	}
	return c
}

func solve() int {
	return combination(40, 20)
}

func main() {
	result := solve()
	fmt.Println(result)
}

File: src/015/p015.java
public class p015 {
    private static long combination(long n, long k) {
        long c = 1;
        for (long i = 0; i < k; i++) {
            c *= n - i;
            c /= i + 1;
        }
        return c;
    }

    private static long solve() {
        return combination(40, 20);
    }

    public static void main(String[] args) {
        long result = solve();
        System.out.println(result);
    }
}
